system/src/commonMain/kotlin/com/splendo/kaluga/system/network/NetworkStateRepo.kt

```diff
@@ -0,0 +1,58 @@
+/*
+ Copyright 2020 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.system.network
+
+import co.touchlab.stately.concurrency.AtomicReference
+import com.splendo.kaluga.state.ColdStateRepo
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import kotlin.coroutines.CoroutineContext
+
+class NetworkStateRepo(
+    context: Any?,
+    coroutineContext: CoroutineContext = Dispatchers.Main
+) : ColdStateRepo<NetworkState>(coroutineContext = coroutineContext) {
+
+    private var _lastKnownNetwork = AtomicReference<Network>(Network.Absent)
+    internal var lastKnownNetwork: Network
```

Comment:
iIs there any value in knowing the last network? If we're not monitoring we don't know if this is correct.

Reply:
If there is it should be a special state or property of initializing (see `Location.Unknown`) as an example.

Reply:
I guess the latter would be pretty cool

Reply:
See this [reply](https://github.com/splendo/kaluga/pull/197#discussion_r521422954)

Reply:
let's do it then 🕶